dockertest/docker_daemon.py

```diff
@@ -159,16 +159,32 @@ def pid():
     if not stdout.startswith('MainPID='):
         raise RuntimeError("Unexpected output from %s: expected MainPID=NNN,"
                            " got '%s'" % (cmd_result.command, stdout))
-    return int(stdout[8:])
-
-
-def cmdline():
+    mainpid = int(stdout[8:])
+    cmd = cmdline(mainpid)
+    if 'dockerd' in cmd[0]:
+        return mainpid
+    # As of April 2017 we may be running dockerd under runc. If so, actual
+    # dockerd process is an immediate child of the one returned by systemd.
+    for child_pid in utils.run("pgrep -P %s" % mainpid).stdout.split():
```

Comment:
A process can go zombie at any time, it's not going to wait for your code between getting the pid and doing something with it.  It's extremely unlikely though, so don't worry about it.

OIC, looks like both ``ps`` and ``pgrep`` come from the ``procps-ng`` package, so no worries.